File: benchmark/c/cpw/n_th_number_whose_sum_of_digits_is_ten/n_th_number_whose_sum_of_digits_is_ten.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int n_th_number_whose_sum_of_digits_is_ten ( int n ) ;
int n_th_number_whose_sum_of_digits_is_ten ( int n ) {
  int count = 0;
  for ( int curr = 1;
  ;
  curr ++ ) {
    int sum = 0;
    for ( int x = curr;
    x > 0;
    x = x / 10 ) sum = sum + x % 10;
    if ( sum == 10 ) count ++;
    if ( count == n ) return curr;
  }
  return - 1;
}

File: translation_result/else_bms/fail/c/cpw/n_th_number_whose_sum_of_digits_is_ten/gemini/CAPR/n_th_number_whose_sum_of_digits_is_ten.rs

fn n_th_number_whose_sum_of_digits_is_ten(n: i32) -> i32 {
    let mut count = 0;
    for curr in 1.. {
        let mut sum = 0;
        let mut x = curr;
        while x > 0 {
            sum += x % 10;
            x /= 10;
        }
        if sum == 10 {
            count += 1;
        }
        if count == n {
            return curr;
        }
    }
    -1
}
